stan-dev/math
# Created top-level includes: {prim, rev, fwd, mix} x {scal, arr, mat}

#### Summary:

Created 12 header files that can be included by users of the math library that will include the template metaprograms in the correct order.

Including `stan/math.hpp` results in the same behavior as before. It includes `stan/math/rev/mat.hpp`.

All tests within each folder includes the appropriate header. This will make testing easier.

#### Intended Effect:

Makes using Stan Math easier. Makes includes easier for testing purposes.


#### How to Verify:

See 09864d1e28fcf68cb18b6a5df8ae9d023d4020cc for the creation of the 12 files and update to `stan/math.hpp`.

I've touched every test file, so it'll be hard to diff, but you'll be able to track at least that first change.

#### Side Effects:

- Makes tests easier. I split apart old tests that had multiple functions inside them (`meta/traits_test.cpp` and `fun/transform_test.cpp`) so they line up with our other tests.
- The includes for `CSR_extract_*` were narrowed down to what they use. (009a2374853e7a327ed8132b9dfa1d8a553dedb3)
- There were two identical definitions of `divide()`; I fixed it by removing one of them. (efa256785fdae353de55301644196b9616ee4231)

#### Documentation:

@bob-carpenter, I can put the usage of these things on the wiki once we're close to merging.

#### Reviewer Suggestions: 

@bob-carpenter
